### DISCUSSION TITLE
Question about upload files

### DISCUSSION THREAD

**Discussion by juanico10** [gollum/gollum#1899] [Q&A]
Hello!

Does anyone know how I can make Gollum show the files I load in the folder?
I mean, I upload a `.md` in a folder and it shows it.

Best regards!

**Comment by benjaminwil** [comment]
Hey there. I am not sure I fully understand your issue. Could you give an example of how you're uploading files, and what kind of files they are?

Gollum will only show files that have been committed to the underlying git repository. So, for example, if you made a copy of an existing file, but did not commit it, it won't show up:

```
cd path-to-my-gollum-wiki
cp ./Home.md ./Home-copy.md

# Home-copy.md will not show up in the wiki yet...

git add ./Home-copy.md && \
  git commit -m "Commit Home-copy.md"

# Now, Home-copy.md will show up in the Gollum "Overview" page
# and be navigable.
```

My example is with Markdown files, but this should be the same for any filetype.

I hope this answers your question. But if it doesn't, I hope your example can help us get closer to an answer.

**Comment by juanico10** [comment]
Hello.

I am uploading the files to my server, via sftp to the local `/wiki` folder of my container. The files are `(.md)`.
Then I go to my website and refresh the browser window and they don't appear.
What do I have to do to make Gollum accept them?

Best regards!

**Reply by bartkamphorst** [reply]
Hi @JuanRodenas ,

Judging from what you're saying, I think the problem is as @benjaminwil suggested, namely that you are not committing your files to the underlying git repository. Uploading the files with `sftp` just puts the files in the working directory, but you have yet to tell `git` (and thus gollum) that those files belong to the relevant repository. 

To commit your files, you should start a regular `ssh` session to your server, and `cd` into the `/wiki` directory. Issuing the command `git status` should show you all the files that are there but have not yet been committed to the repository. 

To commit all the markdown files, run the following:
```bash
git add *.md && git commit -m "Commit my pages"
```
Once the files are committed to the repository, they should show up in your gollum webinterface.

NB. All of this assumes that gollum is indeed initiated with `/wiki` as its path.

**Comment by juanico10** [comment]
Hello!

Thanks a million for the reply. I understand what you mean, I was not aware of this about gollum.
As soon as I can test the result I will confirm it.
Thanks a million for this.

Best regards.

**Comment by dometto** [comment]
@JuanRodenas since you had been using SFTP, maybe a good solution for you could also be to push to your server from your local git repository via ssh, see .e.g [here](https://howto.lintel.in/git-how-to-push-code-to-repository-on-remote-ssh-server).

**Comment by juanico10** [comment]
Hello!!

I tried to test and I got an error.
Where do I have to perform the git, inside the container no?

Best regards
